src/app/action-items/action-items.component.ts

```diff
@@ -1,30 +1,42 @@
-import { ActionTasksElement } from './../services/tasksService/tasks.service';
+import { AddItemComponent } from './../add-item/add-item.component';
 import { Component, OnInit, OnDestroy } from '@angular/core';
 import { map } from 'rxjs/operators';
-import { TasksService } from '../services/tasksService/tasks.service';
+import { TasksService, ActionItem } from '../services/tasksService/tasks.service';
 import { DaysLeftToDeadlineService } from '../services/daysLeftToDeadlineService/days-left-to-deadline.service';
+import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
+import { Subscription } from 'rxjs';
 
-export interface ActionTasksElementMapped extends ActionTasksElement {
-  dueDay: number;
+export interface ActionItemMapped extends ActionItem {
+  dueDay?: number;
 }
-
 @Component({
   selector: 'app-action-items',
   templateUrl: './action-items.component.html',
   styleUrls: ['./action-items.component.css']
 })
-export class ActionItemsComponent implements OnInit {
-  dataSource: ActionTasksElementMapped[];
-  loading = true;
-  currentDate = new Date();
-  constructor(private tasksService: TasksService, private daysCountService: DaysLeftToDeadlineService) {}
+export class ActionItemsComponent implements OnInit, OnDestroy {
+  dataSource: ActionItemMapped[];
+  loading = false;
+  currentDate: Date = new Date();
+  subscription: Subscription;
```

Comment:
**FIX:** This subscription is never assigned. Please removed it or use it. I'd suggest ot use `first` in loading action items.